src/goto-instrument/cover.cpp

```diff
@@ -225,13 +234,26 @@ std::unique_ptr<cover_configt> get_cover_config(
     return {};
   }
 
+  std::string cover_only = options.get_option("cover-only");
   // cover entry point function only
-  if(options.get_bool_option("cover-function-only"))
+  if(cover_only == "function")
   {
     function_filters.add(
       util_make_unique<single_function_filtert>(
         message_handler, config.main_symbol->name));
   }
+  else if(cover_only == "file")
+  {
+    function_filters.add(
+      util_make_unique<file_filtert>(
+        message_handler, config.main_symbol->location.get_file()));
+  }
+  else if(!cover_only.empty())
+  {
+    msg.error() << "Argument to --cover-only not recognised: "
+                << cover_only << messaget::eom;
+    return {};
```

Comment:
inconsistent error handling - should work the same as providing two conflicting cover commands

Reply:
Are you sure you want me to change this? What I've done there is I've used the same syntax as other error messages in the same function.

Reply:
Yeah OK you're probably right - looked weird in the context of the place where you were throwing an exception.